### ISSUE TITLE
Elasticsearch data source with SigV4 auth error

### ISSUE THREAD

**Issue by GustavJaner** [grafana/grafana#31099]
**What happened**:
When upgrading Grafana to v7.3.5 or above (also tried v7.4.0), all the Elasticsearch data sources show `Elasticsearch error: undefined`. The same error is shown in all panels using Elasticsearch and also when trying to "Save & Test" any of the  Elasticsearch data sources.  
Error from the Grafana logs: `error="http: proxy error: invalid SigV4 auth type"`

**Browser**
```
# on "Save & Test" of an ES datasource

error: "Bad Gateway"
message: "Elasticsearch error: undefined"
```

**Logs**
```
# on "Save & Test" of an ES datasource

││ t=2021-02-10T14:07:14+0000 lvl=dbug msg="Received command to update data source" logger=datasources url=https://....
││ t=2021-02-10T14:07:14+0000 lvl=dbug msg="Applying default URL parsing for this data source type" logger=datasource type=elasticsearch url=https://....
││ t=2021-02-10T14:07:15+0000 lvl=dbug msg="Querying for data source via SQL store" logger=datasources id=15 orgId=1
││ t=2021-02-10T14:07:15+0000 lvl=dbug msg="Applying default URL parsing for this data source type" logger=datasource type=elasticsearch url=https://....
││ t=2021-02-10T14:07:15+0000 lvl=eror msg="Data proxy error" logger=data-proxy-log userId=2 orgId=1 uname=.... path=/api/datasources/proxy/15/..../_mapping remote_addr=x.x.x.x referer=https://..../datasources/edit/15/ error="http: proxy error: invalid SigV4 auth type"
││ t=2021-02-10T14:07:15+0000 lvl=eror msg="Request Completed" logger=context userId=2 orgId=1 uname=.... method=GET path=/api/datasources/proxy/15/..../_mapping status=502 remote_addr=x.x.x.x time_ms=11 size=0 referer=https://..../datasources/edit/15/
```

Our Elasticsearch data sources are configured with the **SigV4 auth** option. This was working like a charm when v7.3.3 was released (which included this PR: https://github.com/grafana/grafana/pull/29065, which solved this issue: https://github.com/grafana/grafana/issues/28931). Before v7.3.3, we had to auth ES data sources by using  _aws-sigv4-proxy_ as a sidecar container.  
However, when upgrading to v7.3.5 all the Elasticsearch data sources brake (downgrading to v7.3.3 again solves the issue without altering any other configurations).

**Environment**:
- Grafana version: v7.3.5
- Data source type & version: Elasticsearch v6 & v7
- OS Grafana is installed on: EKS
- User OS & Browser: Firefox & Chrome

**Comment by GustavJaner** [comment]
@wbrowne do you know what changed between the versions?

**Comment by wbrowne** [comment]
@GustavJaner I believe the issue is related to how the default auth type is set on the frontend - this should be fixed in 7.3.5, but looks like it's causing problems in this case.

If you manually update the auth provider field, does that help? The error you see now I believe is due to it being sent as `""`. You could also check the request/response to the Grafana server to verify. It would be interesting to see what the response is for you on 7.3.3 and 7.3.5 for `/api/datasources/<datasourceID>` actually if you're able to provide it (sensitive info scrubbed of course)

**Comment by GustavJaner** [comment]
On point! Updating the auth provider field was enough. Thanks a lot for the quick response!
